vault/src/system.rs

```diff
@@ -364,6 +362,23 @@ impl VaultIdManager {
         Ok(())
     }
 
+    async fn sweep_shared_wallet(&self) -> Result<(), Error> {
+        if self.btc_rpc_shared_wallet.get_pruned_height().await? == 0 {
+            // no need to sweep, full node can rescan
+            return Ok(());
+        } else if self.btc_rpc_shared_wallet_v2.get_last_sweep_height().await?.is_some() {
+            // already has sweep tx
+            return Ok(());
+        }
+
+        // sweep funds from shared wallet to shared-v2
+        let shared_v2_wallet_address = self.btc_rpc_shared_wallet_v2.get_new_sweep_address().await?;
+        let txid = self.btc_rpc_shared_wallet.sweep_funds(shared_v2_wallet_address).await?;
```

Comment:
Should we wait for block inclusion?

Reply:
No I think it should be fine, the wallet will then handle redeem / replace payments automatically using the funds from the sweep tx even if not included.

Reply:
Actually I will await the last sweep tx to prevent a crash loop when importing the issue addresses, which I've designed to fail if the tx is not yet included since otherwise we cannot determine at which height to import issue addresses from.